atomate/common/firetasks/glue_tasks.py

```diff
@@ -73,6 +73,94 @@ def get_calc_loc(target_name, calc_locs):
         return calc_locs[-1]
 
 
+@explicit_serialize
+class GrabFilesFromCalcLoc(FiretaskBase):
+    """
+    Based on CopyVaspOutputs but for general file copying.
+
+    Required params:
+        filenames (str): filenames to copy.
+        name_prepend (str): string to prepend filenames, e.g. can be a directory.
+        name_append (str): string to append to filenames.
+
+    Optional params:
+        calc_dir: directory to copy from.
+        calc_loc: name of fw to get location for.
+    """
+
+    required_params = ["filenames", "name_prepend","name_append"]
+    optional_params = ["calc_dir", "calc_loc"]
+
+    def run_task(self,fw_spec=None):
+
+        if self.get('calc_dir',False):
+            filesystem = None
+        elif self.get('calc_loc',False):
+            calc_loc = get_calc_loc(self.get('calc_loc',False), fw_spec["calc_locs"])
+            calc_dir = calc_loc["path"]
+            filesystem = calc_loc["filesystem"]
+        else:
+            raise ValueError("Must specify either calc_dir or calc_loc!")
+
+        fileclient = FileClient(filesystem=filesystem)
+        calc_dir = fileclient.abspath(calc_dir)
+
+        all_files = fileclient.listdir(calc_dir)
+
+        if self.get('filenames',False):
+            if type(self.get('filenames',False)) == list:
+                files_to_copy = self.get('filenames',False)
+            elif isinstance(self.get('filenames',False), six.string_types):
+                files_to_copy = [ self.get('filenames',False) ]
+            else:
+                ValueError("Must have a list of strings or a strings!")
+        else:
+            raise ValueError("Must have a list of filenames!")
+
+        # determine what files need to be copied
+        if "$ALL" in self.get('filenames',False):
+            files_to_copy = all_files
+
+        # start file copy
+        for f in files_to_copy:
+            prev_path_full = os.path.join(calc_dir, f)
+            # prev_path = os.path.join(os.path.split(calc_dir)[1], f)
+            dest_fname = self.get('name_prepend',"")+f+self.get('name_append',"")
+            dest_path = os.path.join(os.getcwd(), dest_fname)
```

Comment:
dest_path should probably be settable. None=os.getcwd(), otherwise the user can set the dir